curvlinops/inverse.py

```diff
@@ -212,68 +215,255 @@ class KFACInverseLinearOperator(_InverseLinearOperator):
     def __init__(
         self,
         A: KFACLinearOperator,
-        damping: Tuple[float, float] = (0.0, 0.0),
+        damping: Union[float, Tuple[float, float]] = 0.0,
+        use_heuristic_damping: bool = False,
+        min_damping: float = 1e-8,
+        use_exact_damping: bool = False,
         cache: bool = True,
     ):
-        """Store the linear operator whose inverse should be represented.
+        r"""Store the linear operator whose inverse should be represented.
 
         Args:
             A: ``KFACLinearOperator`` whose inverse is formed.
-            damping: Damping values for all input and gradient covariances.
-                Default: ``(0., 0.)``.
+            damping: Damping value(s) for all input and gradient covariances. If tuple,
+                the first value is used for the input covariances and the second value
+                for the gradient covariances. Note that if heuristic or exact damping is
+                used the damping cannot be a tuple. Default: ``0.``.
+            use_heuristic_damping: Whether to use a heuristic damping strategy by
+                `Martens and Grosse, 2015 <https://arxiv.org/abs/1503.05671>`_
+                (Section 6.3). For input covariances
+                :math:`A \in \mathbb{R}^{n \times n}` and gradient covariances
+                :math:`B \in \mathbb{R}^{m \times m}`, we define
+                :math:`\pi := \sqrt{m \frac{\text{tr}(A)}{n \text{tr}(B)}}` and set the
+                damping for the input covariances :math:`A` to
+                :math:`\max(\sqrt{\text{damping}} \pi, \text{min_damping})` and for the
+                gradient covariances :math:`B` to
+                :math:`\max(\sqrt{\text{damping}} / \pi, \text{min_damping})`.
+                Default: ``False``.
+            min_damping: Minimum damping value. Only used if ``use_heuristic_damping``
+                is ``True``. Default: ``1e-8``.
+            use_exact_damping: Whether to use exact damping, i.e. to invert
+                :math:`(A \otimes B) + \text{damping} \mathbf{I}`. This is implemented
+                via eigendecompositions of the Kronecker factors. Default: ``False``.
```

Comment:
Maybe add reference to influence function paper here.

Reply:
Also, can we add `KFACInverseLinearOperator` to the docs? Maybe under **Inverses**?